oo.py

```diff
@@ -0,0 +1,276 @@
+    # 3.0: Integrate with functional api
+    __version = 3.0
+
+    def __init__(self, host="https://app.conduce.com", api_key=None, email=None, password=None, insecure=True):
+        """
+        Configure an API client that can connect to Conduce servers.
+        You may provide either an API key or an email and password.  The API key
+        is strongly encouraged since it does not expose a user's credentials, and
+        does not provide full access to a user's account if compromised.
+        If email/password are provided, you must call the login() function to
+        get a cookie before making API calls.
+
+        :param host: optional host, only used if connecting to beta/preview servers
+        :param api_key: optional api key.  This is the recommended way to use this client
+        :param email: optional email if using login
+        :param password: optional pasword if using login
+        :rtype: ConduceAPIClient
+        """
+        #TODO: Integrate this with config.py
+        if api_key is None:
+            #using email & password
+            if email is None and password is None:
+                raise ValueError("api_key or email & password required")
+            if email is None or password is None:
+                raise ValueError("both email and password must be provided")
+        else:
+            #If an api key is specified, make sure email & password are none
+            if email is not None or password is not None:
+                #TODO: Use the warning lib
+                print "Warning, ignoring email and password inputs because api key provided"
+            email = None
+            password = None
+        #TODO: Make sure API has valid characters letters, numbers & - & _
+        self._api_key = api_key
+        self._email = email
+        self._password = password
+        self._cookies = None
+        self._verify = not insecure
+        info = urlparse.urlparse(host)
+        scheme = info.scheme if info.scheme else "https"
+        hostname = info.hostname if info.hostname else info.path
+        if hostname.endswith("/"):
+            hostname = hostname[:-1]
+        #TODO: Regex check host
+        self._host = "{}://{}".format(scheme, hostname)
+        self._hostname = hostname
+
+    def login(self):
+        if self._api_key:
+            return
+        login_data = json.dumps({'email': self._email, 'password': self._password, 'keep': True})
+        headers = {'Content-Type': 'application/json'}
+        req = requests.post('%s/conduce/api/v1/user/login' % self._host, data=login_data, headers=headers, verify=self._verify)
+        if not req.ok:
+            raise ConduceError(req.url, req)
+        self._cookies = req.cookies
+        #HACK!!!
+        req = self.api_get("/apikeys/list")
+        apikeys = req.json()
+        if len(apikeys) == 0:
+            raise NotImplementedError("Create an API key under the {} user to use the OO API".format(self._email))
```

Comment:
Also, if we can list api keys, we can certainly do the rest of the operations without an api key, can't we?